### PR TITLE
fix: encapsulate request

### PR DESCRIPTION
**Pull request by Ignorant-lxf**

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -36,8 +36,16 @@ func (c *Context) Next() {
 	}
 }
 
-func (c *Context) Request() *fasthttp.Request   { return &c.fc.Request }
-func (c *Context) Response() *fasthttp.Response { return &c.fc.Response }
+func (c *Context) RequestMethodIsOptions() bool { return c.fc.Request.Header.IsOptions() }
```
Comment (Aoang) [inline context.go]: Removing RequestMethod would be intuitive

File: context.go [hunk 2]
```diff
@@ -36,8 +36,16 @@ func (c *Context) Next() {
...
+func (c *Context) RequestMethodIsGet() bool     { return c.fc.Request.Header.IsGet() }
+func (c *Context) RequestMethodIsPost() bool    { return c.fc.Request.Header.IsPost() }
+func (c *Context) RequestMethodIsHead() bool    { return c.fc.Request.Header.IsHead() }
+func (c *Context) RequestMethodIsPut() bool     { return c.fc.Request.Header.IsPut() }
+func (c *Context) RequestMethodIsDelete() bool  { return c.fc.Request.Header.IsDelete() }
+func (c *Context) RequestMethodIsPatch() bool   { return c.fc.Request.Header.IsPatch() }
+
+func (c *Context) PeekHeader(k string) []byte { return c.fc.Request.Header.Peek(k) }
```
Comment (Aoang) [inline context.go]: Header(s string) []byte

File: context.go [hunk 3]
```diff
@@ -36,8 +36,16 @@ func (c *Context) Next() {
...
+func (c *Context) PeekAuthorization() []byte  { return c.PeekHeader("Authorization") }
```
Comment (Aoang) [inline context.go]: a bit redundant?

**Review by Aoang — CHANGES_REQUESTED** [review]

File: context.go (hunk 1 above)
Comment (Ignorant-lxf) [reply context.go]: done

File: context.go (hunk 2 above)
Comment (Ignorant-lxf) [reply context.go]: done

File: context.go (hunk 3 above)
Comment (Ignorant-lxf) [reply context.go]: done

**Review by Aoang — APPROVED** [review]

**Pull request merged**